Scripts/operations.py

```diff
@@ -323,6 +338,91 @@ def check_duplicates(current, config, miscData, allCommentsDict, videoID):
       print(f" Analyzing For Duplicates: [ {scannedCount/authorCount*100:.2f}% ]   (Can be disabled & customized in config)".ljust(75, " "), end="\r")
 
 
+############################# Check Text Duplicates #####################################
+def check_reposts(current, config, miscData, allCommentsDict, videoID):
+  # Get Lenvenshtein Distance Setting
+  try:
+    levenshtein = float(config['levenshtein_distance'])
+    if levenshtein < 0 or levenshtein > 1:
+      print("\nError: Levenshtein_distance config setting must be between 0 and 1. Defaulting to 0.9")
+      input("\nPress Enter to continue...")
+      levenshtein = 0.9
+  except ValueError:
+    print("\nError: Levenshtein_distance config setting must be a number between 0 and 1. Defaulting to 0.9")
+    input("\nPress Enter to continue...")
+    levenshtein = 0.9
+
+  # Get duplicate count setting
+  try:
+    repost_minimum_text_length = int(config['repost_minimum_text_length'])
+    if repost_minimum_text_length < 1:
+      repost_minimum_text_length = 25
+      print("\nError: Repost_minimum_text_length config setting must be greater than 0. Defaulting to 25.")
+      input("\nPress Enter to continue...")
+  except ValueError:
+    repost_minimum_text_length = 25
+    print("\nError: Repost_minimum_text_length config setting is invalid. Defaulting to 25.")
+    input("\nPress Enter to continue...")
+
+  repostCommentsDict = []
+
+  # Calculate number of authors to check, for progress
+  authorCount = len(allCommentsDict)
+  scannedCount = 0
+
+  # Look for multi-author duplicates
+  for authorID, authorCommentsList in allCommentsDict.items():
+    # Don't scan channel owner, current user, or any user in whitelist
+    if auth.CURRENTUSER.id == authorID or miscData.channelOwnerID == authorID or authorID in miscData.resources['Whitelist']['WhitelistContents']:
+      scannedCount +=1
+      print(f" Analyzing For Reposts: [ {scannedCount/authorCount*100:.2f}% ]   (Can be disabled & customized in config)".ljust(75, " "), end="\r")
+    else:
+      for commentDict in authorCommentsList:
+        matched = False
+        for i,x in enumerate(repostCommentsDict):
```

Comment:
This can get slow on 10k+ comment sections as it iterates over an ever growing list.

Reply:
https://github.com/seatgeek/thefuzz could be a good solution